lib/fastly/belongs_to_service_and_version.rb

```diff
@@ -16,9 +16,9 @@ def version
     end
 
     # Get the number of the Version this belongs to
-    def version_number # rubocop:disable all
-      @version
-    end # rubocop:enable all
+    def version_number
+      @version ||= nil
```

Comment:
What's the warning here? An undefined instance variable will already point to `nil`.

```ruby
> @foobar 
=> nil
> @foobar || = nil
> @foobar 
=> nil

Reply:
Also, memoizing an instance var in a public method and not in `initialize` is a _little_ surprising.

Reply:
@andriantoniades could you please post the warning here? We talked about an `initialize` method or doing the memoize, and I don't have a strong preference either way. If using `initialize` is more standard, we can switch to that.

Reply:
Sure! The warning was this:
`warning: instance variable @version not initialized`

Reply:
This is the same style as the two methods above it. If we're choosing a different style, then this is going to be a bigger change as we'd want to change it everywhere. So for now, I'm going to use this to clear the warning and we can have a bigger discussion about memoization in a public method with the broader team.